factory-reset/eos-factory-reset

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/python3
+import argparse
+import sys
+import subprocess
+
+# Stages list
+# Each stage has the form (short option, name, description)
+# In addition from the stages listed bellow, the following stage names and
+# short option names are already taken by other command line arguments:
+# ('q', 'quiet', 'Suppress all output and don\'t ask for confirmation')
+STAGES = [
+    ('u', 'users', 'remove all user accounts (and home directory)'),
+]
+
+def yes_or_no(msg):
+    answer = ''
+    while answer not in ['y', 'n']:
+        answer = input(msg + ' ').lower()
+    return answer == 'y'
+
+if __name__ == '__main__':
+    desc = 'reboot the machine and perform a factory reset.'
+    parser = argparse.ArgumentParser(description=desc,
+                                     argument_default=argparse.SUPPRESS)
+    parser.add_argument('-q', '--quiet', action='store_true')
+
+    group = parser.add_argument_group('factory reset process stages',
+        'select which stages of the factory reset process should be performed '
+        'on the next boot. if no stage is selected all stages will be enabled')
+    for n, name, desc in STAGES:
+        group.add_argument('-'+n, '--'+name, help=desc, action='store_true')
+    args = parser.parse_args()
+
+    quiet = vars(args).pop('quiet', False)
+    stages = list(vars(args).keys())
+
+    if len(stages) == 0:
+        for n, name, desc in STAGES:
+            stages.append(name)
+        if not quiet:
+            print('No stage selected, selecting all stages')
+
+    if not quiet:
+        msg = 'Scheduling a factory reset for the next boot with stages: '
+        msg += ' '.join(stages)
+        msg += '\nDo you want to proceed [y/n]?'
+        if not yes_or_no(msg):
+            sys.exit(0)
+
+    for stage in stages:
+        if (stage):
```

Comment:
no need for ( )

Reply:
In which case this is falsy?

Reply:
@dnohales Good catch, this was leftover my initial draft where I was iterating over the dictionary returned by argparse, and without suppressing unused argumets.